hardware_interface/src/sensor_hardware.cpp

```diff
@@ -54,9 +54,10 @@ hardware_interface_status SensorHardware::get_status() const
   return impl_->get_status();
 }
 
-return_type SensorHardware::read_sensor(std::shared_ptr<components::Sensor> sensors)
+return_type SensorHardware::read_sensors(
```

Comment:
this change is also a little unexpected, I followed the pattern from the interface class but was wondering if this is intended for the parsing PR, feels like a leftover from other stuff

Reply:
This is intended change. The change from `sensors` to `sensor` came from @Karsten1987 just before merging #121. Nevertheless, we agreed that as `SensorHardware` can have multiple `Sensor` types, an example is Camera with integraded IMU sensor. Example URDF for this is in `test_component_parser.cpp` line 395 (Example for "Sensor only").